examples/step-32/step-32.cc

```diff
@@ -1864,16 +1864,16 @@ namespace Step32
       stokes_partitioning.push_back(stokes_index_set.get_view(0, n_u));
       stokes_partitioning.push_back(stokes_index_set.get_view(n_u, n_u + n_p));
 
-      DoFTools::extract_locally_relevant_dofs(stokes_dof_handler,
-                                              stokes_relevant_set);
+      stokes_relevant_set =
+        DoFTools::extract_locally_relevant_dofs(stokes_dof_handler);
```

Comment:
I think in this place the regular RVO will probably not help the change will give rise to (very slightly) slower code than we had before, as we need to call the `operator=` here from the temporary object. This is different from the case in `step-18` you made above where RVO eliminates the copy (on sane compilers).

I think this is justified for a cheap object like `IndexSet` given the interface uniformity it brings us. Furthermore, I believe we should at some point make the `locally_relevant_dofs` a member variable of the `DoFHandler`, given how often we set up and use these objects and how little memory they consume in the grand scheme of things. Then this gets even more clear that we should do it. But I do not think the case is necessarily as low for all functions that *merely fill* something, so we might need to make a case-by-case decision.

Reply:
See also #11896.